mitodl/mitxpro
# Fixed seed data bugs, added products, added deletion command

#### Pre-Flight checklist


#### What are the relevant tickets?
Fixes #425 
Closes #206 
Closes #207 

#### What's this PR do?
- Fixes bug where course/program pages were added under the root page rather than under the course/program index pages
- Fixes bug where seeded objects couldn't be deleted due to delete-protected ecommerce tables
- Adds products & prices to certain seeded objects
- Adds command that allows you to selectively delete all seeded data or just some specific objects

#### How should this be manually tested?
Some of the course run titles in the original `seed_data.json` were poorly set. They were corrected in this PR, which provides an opportunity to test out the delete command. Start by running the following:
- `manage.py delete_seed_data --type courserun --title "SEED Systems Engineering - Fall 2019"`
- `manage.py delete_seed_data --type courserun --title "SEED Systems Engineering - Spring 2020"`
- `manage.py delete_seed_data --type courserun --title "SEED Systems Engineering - Fall 2020"`
- `manage.py delete_seed_data --type courserun --title "SEED Quantitative Methods - August 2019"`

Then run the full seed data command: `manage.py seed_data`

Everything should work, and those course runs should be newly created with correct titles. You should also see that `Product`s/`ProductVersion`s were created or were ignored if they already existed.

You can also test out broader delete functionality:
- Delete all seed data: `manage.py delete_seed_data`
- Delete a program (which deletes all courses, course runs): `manage.py delete_seed_data --type program --title "SEED Digital Learning"`

#### Any background context you want to provide?
- As this code is only related to seed data creation/deletion, feel free to adjust coding standards as you see fit 😁 
- Regarding #207, we already have some CMS page properties in the seed data. If any more properties are desired, we can create specific tickets for them and they can be added easily
- Regarding #206, that issue asks for some coupons to be created, but now that we have the coupon creation and bulk enrollment forms, adding your own coupons is very easy, so it didn't seem necessary to seed any
